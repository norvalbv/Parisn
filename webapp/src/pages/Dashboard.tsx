import Radial from "../components/Design/Radial";

const Dashboard = () => {
  return (
    <div className="min-h-screen">
      <div className="flex items-center bg-green-50">
        {/* <div className="w-full bg-red-100"> hasdasd asd i</div> */}
        <div className="relative bg-red-50 flex-1">
          <Radial />
          <div className="text-white absolute">
            Parisn is a city in the north of France. It is the capital of the
          </div>
        </div>
      </div>
    </div>
  );
};

export default Dashboard;
