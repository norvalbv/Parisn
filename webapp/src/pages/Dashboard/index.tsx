import Radial from '../../components/Design/Radial';
import { DASHBOARD_IMAGE, DASHBOARD_VIDEO } from '../../constants';
import Button from '../../components/Button';
import { ParentSize } from '@visx/responsive';
import { Line } from '@visx/shape';

const Dashboard = () => {
  return (
    <>
      <div className="flex items-center overflow-hidden">
        <div className="h-screen">
          <video
            src={DASHBOARD_VIDEO}
            muted
            autoPlay
            loop
            className="absolute h-[100vh] opacity-50 sepia contrast-125"
          />
          <div className="absolute h-screen bg-fuchsia-200 opacity-10 w-screen" />
        </div>

        <div className="relative flex justify-center items-center flex-1">
          <div className="absolute text-center">
            <div data-testid="PARISN.com">
              <span className="text-[3rem] drop-shadow-[0_0_16px_rgba(255,255,255,0.7)]">
                PARISN
              </span>
              <span className="drop-shadow-[0_0_16px_rgba(255,255,255,0.5)]">.com</span>
            </div>
            <span className="text-3xl tracking-widest block">A Fashion Company</span>
            <Button
              text="View Collections"
              classes="mt-20"
              rounded="lg"
              navigateTo="/collections"
            />
          </div>
        </div>
      </div>
      <div className="h-[150vh] relative overflow-hidden">
        <ParentSize>
          {(parent) => (
            <>
              <svg height={parent.height} width={parent.width} fill="#555555" stroke="#ffffff">
                <Line
                  from={{ x: 0, y: 0 }}
                  to={{ x: parent.width, y: parent.height / 5 }}
                  stroke="#FAFAFA"
                  strokeOpacity={0.75}
                />
                <Line
                  from={{ x: parent.width, y: parent.height / 5 }}
                  to={{ x: 0, y: parent.height / 3 }}
                  stroke="#FAFAFA"
                  strokeOpacity={0.75}
                />
                <Line
                  from={{ x: 0, y: parent.height / 3 }}
                  to={{ x: parent.width, y: parent.height / 1.5 }}
                  stroke="#FAFAFA"
                  strokeOpacity={0.75}
                />
                <Line
                  from={{ x: parent.width, y: parent.height / 1.5 }}
                  to={{ x: 0, y: parent.height }}
                  stroke="#FAFAFA"
                  strokeOpacity={0.75}
                />
              </svg>
              <p
                className={`absolute text-primary-light text-center tracking-widest text-lg rotate-[8.85deg]`}
                style={{ left: parent.width / 2 - 50, top: parent.height / 11.75 }}
              >
                A new way of shopping
              </p>
              <p
                className={`absolute text-primary-light text-center tracking-widest text-lg block rotate-[15deg]`}
                style={{
                  left: parent.width / 2 - 50,
                  top: parent.height / 2.05,
                }}
              >
                Your way, your price
              </p>
              <Radial colour="green" classes="-top-32 -right-32" />
              <Radial colour="purple" classes="-bottom-32 -left-32" />
            </>
          )}
        </ParentSize>
      </div>
      <div className="h-screen flex">
        <p className="flex flex-col justify-center items-center text-center px-40 leading-8">
          <span className="block font-semibold text-inherit underline">PARSIN.com</span>
          Launched in 2022 allows you to choose the price you pay for your items in a reverse
          bidding strategy technique. Starting at £10,000 per item, over time the price of the item
          drops to £0.
        </p>
        <img src={DASHBOARD_IMAGE} alt={DASHBOARD_IMAGE} className="h-screen w-[42.5%]" />
      </div>
    </>
  );
};

export default Dashboard;
