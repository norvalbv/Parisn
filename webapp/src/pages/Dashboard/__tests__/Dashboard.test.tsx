import { FC } from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Dashboard from '../';

describe('<Dashboard />', () => {
  test('Dashboard props with mandatory props', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Dashboard />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
