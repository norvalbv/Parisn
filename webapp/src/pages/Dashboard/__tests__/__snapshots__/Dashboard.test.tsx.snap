// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Dashboard props with mandatory props 1`] = `
[
  <div
    className="flex items-center"
  >
    <img
      alt="http://127.0.0.1:5173/images/dashboard-image.jpg"
      className="h-screen w-[42.5%]"
      src="http://127.0.0.1:5173/images/dashboard-image.jpg"
    />
    <div
      className="relative flex justify-center items-center flex-1"
    >
      <svg
        className="absolute rounded-full opacity-[15%] blur-3xl z-0 "
        style={
          {
            "backgroundColor": "dark",
            "height": "40rem",
            "width": "40rem",
          }
        }
      />
      <div
        className="absolute text-center"
      >
        <div>
          <span
            className="text-[3rem] drop-shadow-[0_0_16px_rgba(255,255,255,0.5)]"
          >
            PARISN
          </span>
          <span
            className="drop-shadow-[0_0_16px_rgba(255,255,255,0.5)]"
          >
            .com
          </span>
        </div>
        <span
          className="text-3xl tracking-widest block"
        >
          A Fashion Company
        </span>
        <button
          className="relative inline-flex items-center justify-center text-center hover:bg-buttons-hover uppercase rounded-lg px-6 py-2.5 text-sm w-full mt-20 border py-4"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "backgroundColor": undefined,
              "color": undefined,
              "width": "25rem",
            }
          }
          type="button"
        >
          <span
            className="inline-block font-semibold"
          >
            View Catalogue
          </span>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="h-[150vh] relative overflow-hidden"
  >
    <div
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <svg
        fill="#555555"
        height={0}
        stroke="#ffffff"
        width={0}
      >
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
      </svg>
      <p
        className="absolute text-primary-light text-center tracking-widest text-lg rotate-[8.85deg]"
        style={
          {
            "left": -50,
            "top": 0,
          }
        }
      >
        A new way of shopping
      </p>
      <p
        className="absolute text-primary-light text-center tracking-widest text-lg block rotate-[15deg]"
        style={
          {
            "left": -50,
            "top": 0,
          }
        }
      >
        Your way, your price
      </p>
      <svg
        className="absolute rounded-full opacity-[15%] blur-3xl z-0 -top-32 -right-32"
        style={
          {
            "backgroundColor": "green",
            "height": "40rem",
            "width": "40rem",
          }
        }
      />
      <svg
        className="absolute rounded-full opacity-[15%] blur-3xl z-0 -bottom-32 -left-32"
        style={
          {
            "backgroundColor": "purple",
            "height": "40rem",
            "width": "40rem",
          }
        }
      />
    </div>
  </div>,
  <div
    className="h-screen flex"
  >
    <p
      className="flex flex-col justify-center items-center text-center px-40 leading-8"
    >
      <span
        className="block font-semibold text-inherit underline"
      >
        PARSIN.com
      </span>
      Launched in 2022 allows you to choose the price you pay for your items in a reverse bidding strategy technique. Starting at £10,000 per item, over time the price of the item drops to £0.
    </p>
    <img
      alt="http://127.0.0.1:5173/images/dashboard-image.jpg"
      className="h-screen w-[42.5%]"
      src="http://127.0.0.1:5173/images/dashboard-image.jpg"
    />
  </div>,
]
`;
